runelite-client/src/main/java/net/runelite/client/plugins/cooking/CookingOverlay.java

```diff
@@ -0,0 +1,111 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.cooking;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.util.stream.IntStream;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.Skill;
+import net.runelite.client.plugins.xptracker.XpTrackerService;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+
+import static net.runelite.api.AnimationID.COOKING_FIRE;
+import static net.runelite.api.AnimationID.COOKING_RANGE;
+
+class CookingOverlay extends Overlay
+{
+	private static final int[] animationIds =
+	{
+		COOKING_FIRE, COOKING_RANGE
+	};
+
+	private final Client client;
+	private final CookingPlugin plugin;
+	private final CookingConfig config;
+	private final CookingSession session;
+	private final XpTrackerService xpTrackerService;
+	private final PanelComponent panelComponent = new PanelComponent();
+
+	@Inject
+	private CookingOverlay(Client client, CookingPlugin plugin, CookingConfig config, CookingSession session, XpTrackerService xpTrackerService)
+	{
+		setPosition(OverlayPosition.TOP_LEFT);
+		this.client = client;
+		this.plugin = plugin;
+		this.config = config;
+		this.session = session;
+		this.xpTrackerService = xpTrackerService;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showCookingStats())
+		{
+			return null;
+		}
+
+		CookingSession session = plugin.getSession();
+
+		panelComponent.setPreferredSize(new Dimension(145, 0));
+		panelComponent.getChildren().clear();
+
+		if (IntStream.of(animationIds).anyMatch(x -> x == client.getLocalPlayer().getAnimation()))
+		{
+			panelComponent.getChildren().add(TitleComponent.builder()
+				.text("Cooking")
+				.color(Color.GREEN)
+				.build());
+		}
+		else
+		{
+			panelComponent.getChildren().add(TitleComponent.builder()
+				.text("NOT cooking")
+				.color(Color.RED)
+				.build());
+		}
+
+		panelComponent.getChildren().add(LineComponent.builder()
+			.left("Cooked:")
+			.right(session.getCookAmount() + (session.getCookAmount() >= 1 ? " (" + xpTrackerService.getActionsHr(Skill.COOKING) + "/hr)" : ""))
```

Comment:
This would throw whenever you aren't cooking.

Reply:
Yeah. I forgot about that part, and I can see that you have provided a better solution for a good amount of the code, so should have been fixed 👍